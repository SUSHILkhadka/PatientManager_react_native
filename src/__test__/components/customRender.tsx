import {NavigationContainer} from '@react-navigation/native';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {render} from '@testing-library/react-native';
import React, {PropsWithChildren} from 'react';
import {Provider} from 'react-redux';

import {allReducers, RootState} from '../../redux_toolkit/stores/store';

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: allReducers,
    preloadedState,
  });
};

export function renderWithProvidersAndNavigation(
  ui: React.ReactElement,
  {preloadedState = {}, store = setupStore(preloadedState), ...renderOptions}: any = {},
) {
  function Wrapper({children}: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <NavigationContainer>{children}</NavigationContainer>
      </Provider>
    );
  }
  return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
