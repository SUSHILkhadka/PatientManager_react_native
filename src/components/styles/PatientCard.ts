import { StyleSheet } from "react-native";
import {COLOR} from "./constants";

const patientCardStyle = StyleSheet.create({
    row: {
      margin: 13,
      padding: 5,
      display: 'flex',
      flexDirection: 'row',
      flexWrap: 'wrap',
      alignItems: 'center',
      borderColor: "white",
      borderWidth: 1.5,
      borderRadius:50,
      backgroundColor:COLOR.black2
    },
    row_texts:{
      position:"relative",
      left: 8,
    },
    row_name:{
      fontSize:16,
      color:COLOR.white1
    },
    row_email:{
      fontSize:12,
      color:COLOR.white2
    },
    image:{
      width: "10%",
      height:35,
      display:"flex",
      alignSelf: "center",
      backgroundColor: COLOR.white2,
      borderRadius:500,
    },
    favouriteIcon: {
      width: "20%",
      borderRadius:500,
      zIndex: 1,
      marginLeft:"auto",
      alignSelf:"flex-end",
    },
    deleteIcon: {
      backgroundColor: COLOR.pink2,
      width: "10%",
      borderRadius:500,
      zIndex: 10,
      marginLeft:30,
      marginRight: 5,
    },
    deleteIcon_text: {
      textAlign: 'center',
      fontSize: 18,
    },
  

  });
  export default patientCardStyle;