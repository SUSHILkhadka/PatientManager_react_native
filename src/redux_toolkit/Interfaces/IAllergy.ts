export interface IAllergy{
    id: number;
    name: string;
    patientId: number;
}
