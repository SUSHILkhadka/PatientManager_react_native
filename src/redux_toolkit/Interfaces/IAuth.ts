type ILoadingStatus={

}
export interface IAuth {
    login: boolean;
    id: number;
    username: string;
    email: string;
    isLoading?: "loading"|"fulfilled"|"failed"
  }