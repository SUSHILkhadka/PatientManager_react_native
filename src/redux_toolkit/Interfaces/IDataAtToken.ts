export interface IDataAtToken {
  name: string;
  id: number;
  email: string;
  expiryDateForRefreshToken: number;
}
