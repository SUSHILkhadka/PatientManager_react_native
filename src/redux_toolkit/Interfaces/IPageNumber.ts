export interface IPageNumber {
  page: number;
  refreshFlag: boolean;
}
