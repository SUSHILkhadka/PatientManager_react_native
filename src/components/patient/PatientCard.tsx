import {AxiosError} from 'axios';
import React, {useState} from 'react';
import {Alert, Button, Image, StyleSheet, Text, View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {IPatient} from '../../redux_toolkit/Interfaces/IPatient';
import {changePage, refreshPage} from '../../redux_toolkit/slices/pageSlice';
import {load} from '../../redux_toolkit/slices/patientSlice';
import {RootState} from '../../redux_toolkit/stores/store';
import {deletePatient} from '../../services/backendCallPatient';
import styleImage from '../styles/Image';
import patientCardStyle from '../styles/PatientCard';
import ToastMessage from '../utils/ToastMessage';

const PatientCard = (props: IPatient) => {
  const pageInfo = useSelector((state: RootState) => state.page);
  const dispatch = useDispatch();
  const [deleteButtonLoading, setDeleteButtonLoading] = useState(false);

  const handleEdit = () => {
    dispatch(load(props));
    dispatch(changePage(3));
  };

  const createDeleteAlertbox = () =>
    Alert.alert('Confirm Delete', 'Are you sure you want to delete this', [
      {
        text: 'Cancel',
        style: 'cancel',
      },
      {text: 'OK', onPress: () => handleDelete()},
    ]);

  const handleDelete = async () => {
    setDeleteButtonLoading(true);
    try {
      const response = await deletePatient(props.patientId);
      dispatch(refreshPage(!pageInfo.refreshFlag));
      ToastMessage(response.message);
    } catch (e: AxiosError | any) {
      ToastMessage(e.response.data.message);
    }
    setDeleteButtonLoading(false);
  };

  return (
    <View style={patientCardStyle.row}>
      <Image
        style={patientCardStyle.image}
        source={{
          uri: props.photoUrl
            ? props.photoUrl
            : 'https://api.minimalavatars.com/avatar/random/png',
        }}
      />
      <Text>{props.name}</Text>
      <Text>{props.email}</Text>
      <Text>{props.address}</Text>
      <Text>{props.specialAttention.toString()}</Text>
      <Text onPress={handleEdit}>Edit</Text>
      <Button
        title="Delete"
        disabled={deleteButtonLoading}
        onPress={createDeleteAlertbox}></Button>
    </View>
  );
};

export default PatientCard;
