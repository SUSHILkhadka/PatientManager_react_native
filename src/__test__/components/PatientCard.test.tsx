import React from 'react';
import {IPatient} from '../../redux_toolkit/Interfaces/IPatient';
import PatientCard from '../../components/patient/PatientCard';
import {renderWithProvidersAndNavigation} from './customRender';
jest.useFakeTimers('legacy');
it('should run patient card', () => {
  const input: IPatient = {
    patientId: 10,
    name: 'First',
    email: 'first',
    contact: 'test',
    dob: 'test',
    address: 'test',
    photoUrl: 'test',
    specialAttention: false,
  };
  const page = renderWithProvidersAndNavigation(<PatientCard {...input} />);

  const deleteIcon = page.getAllByTestId('deleteIcon');
  const favouriteIcon = page.getAllByTestId('favouriteIcon');
  expect(favouriteIcon.length).toBe(1);
  expect(deleteIcon.length).toBe(1);
});
