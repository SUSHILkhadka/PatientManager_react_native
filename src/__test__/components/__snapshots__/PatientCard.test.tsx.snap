// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should run patient card 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#171717",
      "borderColor": "white",
      "borderRadius": 50,
      "borderTopRightRadius": 12,
      "borderWidth": 1.5,
      "display": "flex",
      "flexDirection": "row",
      "height": 200,
      "margin": 6,
      "opacity": 1,
      "position": "relative",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "asdfas",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#171717",
        "borderColor": "white",
        "borderRadius": 500,
        "borderWidth": 1.5,
        "display": "flex",
        "height": "80%",
        "margin": 8,
        "width": "40%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-around",
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      first patient
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      asdfas
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      asdfas
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      asdfas
    </Text>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="favouriteIcon"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "alignSelf": "center",
              "color": "#e51c83",
              "fontSize": 34,
              "marginHorizontal": 10,
            },
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰓒
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e51c83",
        "borderRadius": 500,
        "marginLeft": "90%",
        "opacity": 1,
        "position": "absolute",
        "width": 10,
        "zIndex": 10,
      }
    }
    testID="deleteIcon"
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      ╳
    </Text>
  </View>
</View>
`;
